store/postgres/src/store.rs

```diff
@@ -1051,6 +1062,35 @@ impl StoreTrait for Store {
             );
         }
     }
+
+    fn block_number(
+        &self,
+        _: &SubgraphDeploymentId,
+        hash: H256,
+    ) -> Result<Option<BlockNumber>, StoreError> {
+        use crate::db_schema::ethereum_blocks::dsl;
+
+        let hash = format!("{:x}", hash);
+
+        // We should also really check that the block with the given hash is
+        // on the chain starting at the subgraph's current head. That check is
+        // very expensive though with the data structures we have currently
+        // available. Ideally, we'd have the last REORG_THRESHOLD blocks in
+        // memory so that we can check against them, and then mark in the
+        // database the blocks on the main chain that we consider final
```

Comment:
#1383 will make this easier and cheaper. We'll be able to query the blocks for a network as entities and filter by `isOmmer: false` to only match blocks that are _truly_ on the official chain.

Reply:
For now I suggest we leave this comment in. How we access blocks for a network will change and that will take us back to this comment eventually.

Reply:
Yes, that was my intent: that soon we'll make this comment go away, but doing so is outside the scope of time-travel queries, since it requires _a lot_ of heavy lifting in other places.